src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticsearchIndexDocumentManager.cs

```diff
@@ -0,0 +1,310 @@
+using Elastic.Clients.Elasticsearch;
+using Elastic.Clients.Elasticsearch.Fluent;
+using Elastic.Clients.Elasticsearch.IndexManagement;
+using Elastic.Clients.Elasticsearch.Mapping;
+using Elastic.Clients.Elasticsearch.QueryDsl;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Contents.Indexing;
+using OrchardCore.Entities;
+using OrchardCore.Indexing;
+using OrchardCore.Indexing.Models;
+using OrchardCore.Search.Elasticsearch.Core.Models;
+
+namespace OrchardCore.Search.Elasticsearch.Core.Services;
+
+public sealed class ElasticsearchIndexDocumentManager : IIndexDocumentManager
```

Comment:
We'll find another name

Reply:
I changed it to `IDocumentIndexManager` feel free to suggest something better.